api/v1alpha1/dnshealthcheckprobe_types.go

```diff
@@ -97,14 +97,25 @@ type DNSHealthCheckProbeList struct {
 	Items           []DNSHealthCheckProbe `json:"items"`
 }
 
+func (p *DNSHealthCheckProbe) GetIPAddresses() []string {
+	if net.ParseIP(p.Spec.Address) != nil {
+		return []string{
+			p.Spec.Address,
+		}
+	}
+	ips, _ := net.LookupHost(p.Spec.Address)
+	return ips
+
+}
+
 func (p *DNSHealthCheckProbe) Default() {
 	if p.Spec.Protocol == "" {
 		p.Spec.Protocol = HttpProtocol
 	}
 }
 
 func (p *DNSHealthCheckProbe) ToString() string {
-	return fmt.Sprintf("%v://%v:%v/%v", p.Spec.Protocol, p.Spec.Hostname, p.Spec.Port, p.Spec.Path)
+	return fmt.Sprintf("%v://%v:%v%v", p.Spec.Protocol, p.Spec.Hostname, *p.Spec.Port, p.Spec.Path)
```

Comment:
`p.Spec.Port` could nil pointer panic here right?

Reply:
It could be. I am not sure why we need a port to be a pointer - we only read value anyway. And it being a pointer makes kubebuilder validation useless. Changed it to just int